pkg/driver/driver.go

```diff
@@ -487,7 +487,18 @@ func (d Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 	}
 	volumeSize := data.NewKibiByte(data.KiB * data.ByteSize(requiredKiB))
 
-	volId := d.Storage.CompatibleVolumeId(req.GetName())
+	params, err := volume.NewParameters(req.GetParameters())
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "failed to parse parameters: %v", err)
+	}
+
+	var pvcNamespace, pvcName string
+	if params.UsePvcName {
+		pvcName = req.GetParameters()[ParameterCsiPvcName]
```

Comment:
the values for the keys for sure exist? (and are "" or some defined good value)?

Reply:
They don't have to exist. But that should then just default to empty strings, in which case they get ignored again :shrug:

Reply:
that would be fine, I was just worried that it might access keys that don't exist at all